File: src/main/com/xuechong/bootstraptests/service/TopicService.java
package com.xuechong.bootstraptests.service;

import java.util.List;

import org.hibernate.SessionFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.xuechong.bootstraptests.model.Topic;

@Service
@Transactional
public interface TopicService {
	public void add(Topic t);
	public List<Topic> list(Integer page);
	public int totalPage();
	public void remove(String id);
	public Topic findById(String id);
	public List<Topic> findTop(int top);
	public void setSessionFactory(SessionFactory sessionFactory);
}

File: src/py/TopicService.py

import com.xuechong.bootstraptests.service.TopicService
from org.apache.log4j import Logger
from DaoSupport import DaoSupport

class TopicService(com.xuechong.bootstraptests.service.TopicService,DaoSupport):
    
    logger = Logger.getLogger("TopicService py Impl")
    startIndex = lambda x : x > 0 and (x-1)*6 or 0
    
    def __init__(self):
        """public TopicServie()"""
        if self.logger.isInfoEnabled():
            self.logger.info(str(type(self)) + "init ")
        print str(type(self)) + "init "
        
    def add(self,topic):
        self.getSession().save(topic)
        print "py add"
        
    def list(self,page):
        hql = "FROM Topic ORDER BY createDate DESC"
        query = self.getSession().createQuery(hql)
        query.setFirstResult(self.getIndex(page))
        query.setMaxResults(6);
        return query.list()
    
    def totalPage(self):
        print "py totalPage"
        return 100
    
    def remove(self,id):
        print "py remove " + id
        
    def findById(id):
        print "py find by id"
        return None
    
    def findTop(top):
        print "py find top" + str(top)
        return []
    
